src/main/java/vg/civcraft/mc/civmodcore/serialization/NBTSerialization.java

```diff
@@ -0,0 +1,160 @@
+package vg.civcraft.mc.civmodcore.serialization;
+
+import com.google.common.base.Strings;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.Map;
+import vg.civcraft.mc.civmodcore.api.NBTCompound;
+
+/**
+ *
+ * */
+public final class NBTSerialization {
+
+	private static final String NBT_CLASS_PATH_KEY = "==";
+    private static final Map<String, Class<? extends NBTSerializable>> registeredClasses = new HashMap<>();
+
+    /**
+	 * Registers an {@link NBTSerializable} class and any aliases it may have.
+	 *
+	 * @param clazz The class of the NBTSerializable class.
+	 * @param aliases The alias class name of the class used for when the class has migrated and legacy support is
+	 * required.
+	 *
+	 * @throws IllegalArgumentException Throws if 1) the NBTSerializable class is null, 2) the class does not have a
+	 * public zero argument constructor, or 3) is already registered. (Aliases that are already registered will print a
+	 * warning in lieu of throwing.)
+	 * */
+    public static <T extends NBTSerializable> void registerNBTSerializable(Class<T> clazz, String... aliases) {
+        if (clazz == null) {
+            throw new IllegalArgumentException("Cannot register NBTSerializable: " +
+                    "the given class is null.");
+        }
+        boolean hasValidConstructor = false;
+		for (Constructor<?> constructor : clazz.getDeclaredConstructors()) {
```

Comment:
probably just format the whole file

Reply:
Yeah, I'm really not sure what's going on here... this file is new and the only copy pastes were from within the file itself, which is how some of the wrong documentation came about... weird